apollo-router/src/plugins/rhai/tests.rs

```diff
@@ -38,6 +40,78 @@ use crate::services::SupergraphRequest;
 use crate::services::SupergraphResponse;
 use crate::Context;
 
+// There is a lot of repetition in these tests, so I've tried to reduce that with these two
+// functions. The repetition could probably be reduced further, but ...
+async fn call_rhai_function(fn_name: &str) -> Result<(), Box<rhai::EvalAltResult>> {
+    let dyn_plugin: Box<dyn DynPlugin> = crate::plugin::plugins()
+        .find(|factory| factory.name == "apollo.rhai")
+        .expect("Plugin not found")
+        .create_instance_without_schema(
+            &Value::from_str(
+                r#"{"scripts":"tests/fixtures", "main":"request_response_test.rhai"}"#,
+            )
+            .unwrap(),
+        )
+        .await
+        .unwrap();
+
+    // Downcast our generic plugin. We know it must be Rhai
+    let it: &dyn std::any::Any = dyn_plugin.as_any();
+    let rhai_instance: &Rhai = it.downcast_ref::<Rhai>().expect("downcast");
+
+    let block = rhai_instance.block.load();
+
+    // Get a scope to use for our test
+    let scope = block.scope.clone();
+
+    let mut guard = scope.lock().unwrap();
+
+    // We must wrap our canned response in Arc<Mutex<Option<>>> to keep the rhai runtime
+    // happy
+    let response = Arc::new(Mutex::new(Some(subgraph::Response::fake_builder().build())));
+
+    // Call our rhai test function. If it doesn't return an error, the test failed.
+    block
+        .engine
+        .call_fn(&mut guard, &block.ast, fn_name, (response,))
+}
+
+async fn call_rhai_function_with_arg<'a, T: Sync + Send + 'a + 'static>(
+    fn_name: &'a str,
```

Comment:
We don't need the `'a` lifetime.
```suggestion
async fn call_rhai_function_with_arg<T: Sync + Send + 'static>(
    fn_name: &str,
```

Reply:
@IvanGoncharov Are you still planning on addressing this feedback?

Reply:
Missed that one.
Done ✅